ports/stm32/timer.c

```diff
@@ -314,6 +314,35 @@ STATIC uint32_t compute_prescaler_period_from_freq(pyb_timer_obj_t *self, mp_obj
     return (prescaler - 1) & 0xffff;
 }
 
+// computes prescaler and period so TIM triggers with a period of t_num/t_den seconds
+STATIC uint32_t compute_prescaler_period_from_t(pyb_timer_obj_t *self, int32_t t_num, int32_t t_den, uint32_t *period_out) {
+    uint32_t source_freq = timer_get_source_freq(self->tim_id);
+    if (t_num <= 0 || t_den <= 0) {
+        mp_raise_ValueError("must have positive freq");
+    }
+    uint64_t period = (uint64_t)source_freq * (uint64_t)t_num / (uint64_t)t_den;
+    uint32_t prescaler = 1;
+    while (period > TIMER_CNT_MASK(self)) {
+        // if we can divide exactly, and without prescaler overflow, do that first
+        if (prescaler <= 13107 && period % 5 == 0) {
+            prescaler *= 5;
+            period /= 5;
+        } else if (prescaler <= 21845 && period % 3 == 0) {
+            prescaler *= 3;
+            period /= 3;
+        } else {
+            // may not divide exactly, but loses minimal precision
+            prescaler <<= 1;
+            period >>= 1;
+            if (prescaler > 0x10000) {
```

Comment:
Thanks for the tip.  Yes you are right and I agree it would be good to give the best accuracy here.  I've update the PR with a change.  It's very hard to find input numbers which trigger this case, but one of them was: `pyb.Timer(1, period=1641013233, period_hz=168_000_000//2)` (TIM1 source frequency is 168MHz).